### PR TITLE
Testing Calculator Logic and Components

### PR DESCRIPTION
**Pull request by sharon-odhiambo**
- In this PR we used Jest and React-test-library to perform test for the calculator logics and the website components. 

Highlights:

- Setup React Testing Library
- Test operation and calculation functionalities
- Use snapshot testing to check the UIs

**Cheers and Happy Reviewing!**

### CODE REVIEW

File: src/_tests_/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import calculate from '../../logic/calculate';
```
Comment (FrederickMih) [inline src/_tests_/logic/calculate.test.js:1]: - [ ] Based on the requirements of this project, all the functionalities, and scenarios of the calculate.js module should be covered using jest. Hence let's kindly consider the scenarios where the button name(buttonName) is equal to zero('0'), not zero('!0'), dot(.), plus/minus('+/-').

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @sharon-odhiambo  , @Aquinasmoriasi  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights :green_apple: 

- All tests cases are passing :heavy_check_mark: 
- GitHub workflow has been followed :heavy_check_mark: 
- No Linter issue detected :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 

## Required Changes ♻️ :red_circle: 
- [ ] This is a collaborative project hence let's include the other author in the README.md file. 
- [ ] Based on the requirements of this project,  there should be the usage of jest snapshots for testing all the components. Right now there's only a single snapshot test while you have four components. Let's kindly fix this issue.
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/_tests_/logic/calculate.test.js (hunk 1 above)
Comment (Aquinasmoriasi) [reply src/_tests_/logic/calculate.test.js:1]: Thank you for the review.

**Review by gemmen29 — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 



## Optional suggestions

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Aquinasmoriasi** [comment]
Thank you.

**Pull request merged**